common/src/addressing.rs

```diff
@@ -0,0 +1,101 @@
+// Copyright 2018 Bitwise IO
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crypto::digest::Digest;
+use crypto::sha2::Sha256;
+
+const FAMILY_NAMESPACE: &str = "bond namespace";
+const ORGANIZATION: &str = "organization";
+const PARTICIPANT: &str = "participant";
+const BOND: &str = "bond";
+const HOLDING: &str = "holding";
+const SETTLEMENT: &str = "settlement";
+const RECEIPT: &str = "receipt";
+const ORDER: &str = "order";
+const QUOTE: &str = "quote";
+
+pub fn get_prefix(object: &str) -> String {
```

Comment:
I'm a little confused to why we need both functions `get_prefix` and `hash`? It seems to me you could get the family namespace and by calling `hash(FAMILY_NAMESPACE, 6)` for example. Or otherwise you could have the `get_prefix` function call `hash`.